src/data/repositories/GlassATCDefaultRepository.ts

```diff
@@ -64,14 +65,36 @@ export class GlassATCDefaultRepository implements GlassATCRepository {
     getListOfAtcVersionsByKeys(atcVersionKeys: string[]): FutureData<ListGlassATCVersions> {
         return Future.joinObj(
             atcVersionKeys.reduce((acc, atcVersionKey) => {
-                return {
-                    ...acc,
-                    [atcVersionKey]: this.getAtcVersion(atcVersionKey),
-                };
+                if (validateAtcVersion(atcVersionKey)) {
+                    return {
+                        ...acc,
+                        [atcVersionKey]: this.getAtcVersion(atcVersionKey),
+                    };
+                }
+                console.error(`ATC version key not valid: ${atcVersionKey}`);
+                return acc;
             }, {})
         );
     }
 
+    getRecalculateDataInfo(): FutureData<GlassATCRecalculateDataInfo | undefined> {
+        return this.dataStoreClient.getObject(DataStoreKeys.AMC_RECALCULATION);
+    }
+
+    disableRecalculations(): FutureData<void> {
+        return this.getRecalculateDataInfo().flatMap(recalculateDataInfo => {
+            if (recalculateDataInfo) {
```

Comment:
[subjective] Not that this is wrong, but typically we use a guard here:  `if(!recalculateDateInfo) return Future.success(undefined);`, and we save an indentation level.